### PR TITLE
Introduce Prometheus Metrics regarding CPU and Memory Limits at Cluster Level

### LINKED ISSUE #928: Introduce Prometheus Metrics regarding CPU and Memory Limits and Storage at Cluster Level
As [Sebastian](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#sebastian---tech-lead) and [Mathew](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#matthew---cloud-native-software-engineer) I would like to be able to see Prometheus Metrics regarding CPU Limits and Requests at cluster level so that I can make informed decisions about possible resource problems.
**Acceptance Criteria**

- [x] Follow Project DOD
- [x] We use prometheus as a source for the metrics
- [x] The feature is enabled when prometheus is available in the cluster and disabled when prometheus is disabled.
- [x] We will follow Fiori Guidelines when comes to the style of the graph
- [x] Our users are used to Grafana dashboards, so  when choosing a graph style use https://grafana.com/docs/grafana/latest/visualizations/ as inspiration
- [x] We show CPU,Memory and Storage Limits and Requests
- [x] We can consider using a design that is similar to the one attached - horizontal stacked bars
- [x] This graph will be on the same level 0 separate card to Historical CPU/Memory graph on Cluster Overview
- [x] The Refresh rate of the data is predetermined and user cannot change it
- [x] Fe follow the DOD

**Tasks**

- [ ] task 1

image
image

** External Links **
- https://learnk8s.io/setting-cpu-memory-limits-requests
- https://sysdig.com/blog/kubernetes-limits-requests/
- https://shipit.dev/posts/wasting-money-with-kubernetes.html

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Added a new metrics panel. I went with 1/1 instead of 2/1, as with 2/1 it was too small.
- I didn't find a good query for storage, so it's only CPU & memory.

**Related issue(s)**

Closes #928

### CODE REVIEW

**Review by dariadomagala — COMMENTED** [review]
It doesn't look like the capacity reaches 100%
Screen Shot 2022-05-30 at 08 35 57


Can we have the warning at the end in a different color than the first part of the graph? Also can we mark it somehow that there is an additional message (icon maybe)?
Screen Shot 2022-05-30 at 08 37 05

**Review by dariadomagala — APPROVED** [review]

File: core/src/i18n/en.yaml [hunk 1]
```diff
@@ -1254,6 +1254,14 @@ graphs:
   network-up: Upload
   nodes: Nodes
   pvc-usage: Volume Space Usage
+  resource-commitment:
+    capacity: Capacity
+    cpu: CPU Commitment
+    limits: Limits
+    limits-exceed: Resource limits exceed the 150% of cluster capacity.
```
Comment (grego952) [inline core/src/i18n/en.yaml:1261]:
```suggestion
    limits-exceed: Resource limits exceed 150% of the cluster capacity.
```

File: core/src/i18n/en.yaml [hunk 2]
```diff
@@ -1254,6 +1254,14 @@ graphs:
   network-up: Upload
   nodes: Nodes
   pvc-usage: Volume Space Usage
+  resource-commitment:
+    capacity: Capacity
+    cpu: CPU Commitment
+    limits: Limits
+    limits-exceed: Resource limits exceed 150% of the cluster capacity.
+    memory: Memory Commmitment
```
Comment (grego952) [inline core/src/i18n/en.yaml:1262]:
```suggestion
    memory: Memory Commitment
```

**Review by grego952 — APPROVED** [review]

**Review by dariadomagala — APPROVED** [review]

**Pull request merged**